ui/app/components/job-page/parts/recent-allocations.js

```diff
@@ -0,0 +1,24 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import PromiseArray from 'nomad-ui/utils/classes/promise-array';
+
+export default Component.extend({
+  sortProperty: 'modifyIndex',
+  sortDescending: true,
+  sortedAllocations: computed('job.allocations.@each.modifyIndex', function() {
+    return new PromiseArray({
+      promise: this.get('job.allocations').then(allocations =>
+        allocations
+          .sortBy('modifyIndex')
+          .reverse()
+          .slice(0, 5)
+      ),
+    });
+  }),
```

Comment:
Just about of interest, what's the reason for the PromiseArray? I'm guessing the allocations might not be loaded yet when you ask to sort them or something? If you have a sec would be good to get an understanding, no prob if not.

I had a quick look at some other sorting you are doing and none of them use this, are allocations different or is this something you are going to be moving to across the board?

Reply:
Actually just noticed also, looks like job is an attribute on the component `(job="job")` and then you are reaching up into the job to get the allocations. Just an offshoot, would an alternative be `allocations={{job.allocations}}` so you are sending the data down? Not sure if that is workable for you or not

Reply:
If it isn't clear, the `PromiseArray` object makes it so the returned value always has an array-like interface before and after the promise resolves.

This isn't important elsewhere because `RecordArray` (either from `store.findAll` or `model.get('asyncHasMany')` are used in templates.

Here, additional transformations need to happen (sortBy, reverse, slice). 

If the promise isn't resolved yet, this breaks because it's not an array yet. Simply doing `this.get('job.allocations').then(allocations => allocations.transformations)` also doesn't work because the computed property now returns a promise. The template can't do anything with the promise and the computed property doesn't update when the promise resolves.

By using a PromiseArray, I can do the promise control flow while also ensuring an Array interface as well as a template re-render when the promise resolves.

Reply:
I could use `allocations=job.allocations`, and that may be better. Specifically because of the principle of least authority. I chose to use `job=job` to make all the various `job-page/parts` components feel similar. 

I don't think this violates data-down actions-up because it isn't reaching up into a higher viewer (e.g., `allocations: alias('parentView.job.allocations')`). Instead, a larger than necessary piece of data is being provided to the component and the component is reaching _down_ in that data. 

I'm willing to talk about this more if you wish. I know it's pretty nuanced.

Reply:
Hey, thanks for the further detail.

> I'm willing to talk about this more if you wish. I know it's pretty nuanced.

It's fine I'm not fussed, just thought I'd highlight a alternative approach if it helps. If you decide to go that route at some point and you want to chat then quite happy to, no prob if not.

Back to the PR - Ping me when you tweak the search thing so you can undo your search when there are no results, and I can give it another once over.

Thanks,